src/components/calcite-rating/calcite-rating.tsx

```diff
@@ -0,0 +1,216 @@
+import { Component, Element, h, Host, Prop } from "@stencil/core";
+import { getKey } from "../../utils/key";
+
+@Component({
+  tag: "calcite-rating",
+  styleUrl: "calcite-rating.scss",
+  shadow: true
+})
+export class CalciteRating {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteRatingElement;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  /** specify the theme of scrim, defaults to light */
+  @Prop({ reflect: true }) theme: "light" | "dark";
+
+  /** specify the icon used for the rating, defaults to star */
+  @Prop({ reflect: true }) iconType: "star" | "circle" = "star";
```

Comment:
Right now it only works with the icons that have reciprocal "-f" versions... For now circle and star. 

In places where we lock down icon selections to a few (accordion, split button), 'iconType' has been used to suggest that not any icon should used. Also why it has to be validated since I'm using the filled version of icons.

Could certainly look at allowing any icons and just styling with color but since all the icons are outlined now it's not really clear what's selected 😟.

Reply:
Kind of indifferent on that ... not sure there will be more added anytime soon since they just nixed basically all the filled icons 😆, although if someone does have a need for banana ratings they would probably be able to create a `banana-f` icon... 

These two seem reasonable for rating UI for now but not sure why we wouldn't document the two choices 🤷‍♂️ ...

Reply:
We could also just remove this choice and limit to only use stars for now until someone asks for circles or another icon explicitly...